pkg/storage/common/common.go

```diff
@@ -971,14 +971,36 @@ type DedupeTaskGenerator struct {
 	and generating a task for each unprocessed one*/
 	lastDigests []godigest.Digest
 	done        bool
+	repos       []string // list of repos on which we run dedupe
 	Log         zlog.Logger
 }
 
 func (gen *DedupeTaskGenerator) Next() (scheduler.Task, error) {
 	var err error
 
+	/* at first run get from storage currently found repositories so that we skip the ones that gets synced/uploaded
+	while this generator runs, there are deduped/restored inline, no need to run dedupe/restore again */
+	if len(gen.repos) == 0 {
+		gen.repos, err = gen.ImgStore.GetRepositories()
+		if err != nil {
+			//nolint: dupword
+			gen.Log.Error().Err(err).Msg("dedupe rebuild: unable to to get list of repositories")
+
+			return nil, err
+		}
+	}
+
+	if len(gen.repos) == 0 {
```

Comment:
Bth `if`s have the same condition, Can we make it a single if?

Reply:
I still need to check if there are still no repositories after running GetRepositories(), I moved it inside the first if.

Reply:
Understood. You should also set an empty slice as value for repos in the Reset() method